### PR TITLE
fixes bug 1269453 - Improve cachability for the first signature dates

### PR DESCRIPTION
**Pull request by peterbe**
I put in a print statement [here](https://github.com/peterbe/socorro/blob/6e89e6ac4dab66c622e1c229780c5f4312d11a5b/socorro/external/postgresql/signature_first_date.py#L19) that said `print "Querying for", len(params['signatures']), "signatures"`. 
Then I opened the Top Crashers for each version of Firefox. Before this patch, it said:
```
Querying for 50 signatures
Querying for 50 signatures
Querying for 50 signatures
Querying for 50 signatures
```
Then, after writing this patch, same thing and this time it printed this out:
```
Querying for 50 signatures
Querying for 47 signatures
Querying for 29 signatures
Querying for 40 signatures
```
In other words, it stops querying PG for signatures' first date if it's already done that for some other version. 

So this patch makes it less SQL select work on PG but that's the small part. 
What really makes a difference is that now EACH signature has its own first_date cache and it's 24 hours. It means that we'll only really need something like a total of 150ish queries against PG in a day.  Instead of 200x24. 

That's going to make the Top Crashers page faster.

### CODE REVIEW

File: webapp-django/crashstats/crashstats/models.py [hunk 1]
```diff
@@ -1368,6 +1377,51 @@ class SignatureFirstDate(SocorroMiddleware):
...
+                # remember the cache keys of those we need to look up
+                missing[signature] = cache_key
+
+        if missing:
+            hits = super(SignatureFirstDate, self).get(
+                signatures=missing.keys()
+            )['hits']
+
+            for hit in hits:
+                signature = hit.pop('signature')
```
Comment (adngdb) [inline webapp-django/crashstats/crashstats/models.py]: What's the point of using ``pop`` here? Is there any performance gain compared to ``get``? I can't see any difference locally, and ``pop`` is more confusing in my opinion.

Comment (adngdb) [inline webapp-django/crashstats/crashstats/models.py]: Damn, I should read the rest of the code before commenting! Forget this! :)

**Comment by adngdb** [comment]
r+
